library/psa_crypto.c

```diff
@@ -1011,6 +1013,16 @@ psa_status_t psa_wipe_key_slot( psa_key_slot_t *slot )
      * been wiped. Clear remaining metadata. We can call memset and not
      * zeroize because the metadata is not particularly sensitive. */
     memset( slot, 0, sizeof( *slot ) );
+
+    do
+    {
+        rnd_status = psa_generate_random( (uint8_t *) &slot->key_id,
+                                          sizeof(slot->key_id) );
```

Comment:
How does setting a random key id help, compared to setting to some fixed value which is guaranteed not to be a valid key id?

Reply:
Why should we not set a valid ID? This random value just shouldn't be the one that's marked as the wrong, initial, "slot unused" one. 
Random key ID makes it so that the ID generator is stateless, and the generated ID's have a larger hamming distance from each other.
This might not be inside the thread model for PSA (fault injection) however.
Cons - possible ID clash for two identical IDs. Even then, it's an issue if we generate the same number twice on the same slot, not if we generate two same IDs in two different slots.